packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -14,9 +14,36 @@ export class GOVUKFrontendComponent {
    * Constructs a new component, validating that GOV.UK Frontend is supported
    *
    * @internal
+   * @param {Element | null} [$module] - HTML element to use for component
    */
-  constructor() {
+  constructor($module) {
     this.checkSupport()
+    this.checkInitialised($module)
+
+    const moduleName = /** @type {ChildClassConstructor} */ (this.constructor)
+      .moduleName
+
+    if (typeof moduleName === 'string') {
+      moduleName && $module?.setAttribute(`data-${moduleName}-init`, 'true')
```

Comment:
**suggestion** We can keep the HTML slightly lighter by setting that attribute to `''` rather than `true` (and checking with `hasAttribute` which saves some `!!` casting, as described in the previous comment.).

```suggestion
      moduleName && $module?.setAttribute(`data-${moduleName}-init`, '')
```